### PR TITLE
Add support for download xhr headers

### LINKED ISSUE #374: http headers when download is true
I may have missed this from the docs, but if I want to set an Authorization header on the request when streaming a file, is there a way to do so?

eg:
```
Papa.parse( 'some/endpoint.csv', {
	download: true,
	//downloadConfig: {headers:{authorization:'some token'}} was thinking an option like this may exist but haven't found reference to anything
	step: data => {
		console.log( data );
	},
	complete: this._subject.onCompleted
} );
```

### PR DESCRIPTION
**Pull request by j--w**
Fixes #374

As discussed in issue, tests for NetworkStream require additional infrastructure to be addressed in a new issue.

I could not find any documentation about whether the minified file should be generated in this commit or if there is another process for that. Please advise.

### CODE REVIEW

File: papaparse.js [hunk 1]
```diff
@@ -582,6 +582,19 @@
 
 			xhr.open('GET', this._input, !IS_WORKER);
 
+			if (this._config.downloadRequestHeaders)
```
Comment (pokoli) [inline papaparse.js]: I find it more logical to set it after withCredentials as we are setting up the request there.

File: papaparse.js [hunk 2]
```diff
@@ -582,6 +582,19 @@
 
 			xhr.open('GET', this._input, !IS_WORKER);
 
+			if (this._config.downloadRequestHeaders) 
+			{
+				var headers = this._config.downloadRequestHeaders;
+
+				for (var headerName in headers) 
+				{
+					if (Object.prototype.hasOwnProperty.call(headers, headerName))
```
Comment (pokoli) [inline papaparse.js]: why is this required? Is not enought by setting the setRequestHeader?

File: papaparse.js (hunk 1 above)
Comment (j--w) [inline papaparse.js]: The xhr state must be `OPENED` to set headers.

File: papaparse.js (hunk 2 above)
Comment (j--w) [inline papaparse.js]: Good point, I've removed this redundant check

File: papaparse.js [hunk 3]
```diff
@@ -582,6 +582,16 @@
 
 			xhr.open('GET', this._input, !IS_WORKER);
 
+			if (this._config.downloadRequestHeaders)
```
Comment (pokoli) [inline papaparse.js]: Can we add a comment to indicate that the state must be OPEN to set the request?

Comment (pokoli) [inline papaparse.js]: s/request/header/ sorry :confused:

Comment (j--w) [inline papaparse.js]: No worries, added in latest changes

**Comment by pokoli** [comment]
Code have been merged. Thanks for contributing!
